docs/source/graph-construction/gs-processing/gpartition/ec2-clusters.rst

```diff
@@ -0,0 +1,122 @@
+======================================
+Running partition jobs on EC2 Clusters
+======================================
+
+Once the :ref:`distributed processing setup<gsprocessing_distributed_setup>` is complete,
+you can start the partition jobs. In summary this doc will provide instructions on how to setup the EC2 cluster and
+start GPartition jobs on EC2 cluster.
+
+Create a GraphStorm Cluster
+----------------------------
+
+Setup the instance of a cluster
+................................
+A cluster contains several instances each of which can run GraphStorm Docker container.
+
+To create such a cluster please follow the :ref:`Environment Setup <setup_docker>`. The guide shows you how to build GraphStorm Docker images, and use a Docker container registry, e.g. AWS ECR, to upload the GraphStorm image to an ECR repository and pull it on the instances in the cluster.
+
+If you can't access a Docker registry from your environment, in **each** instance of the cluster, follow the :ref:`Environment Setup <setup_docker>` description to build a GraphStorm Docker image, and start the image as a container. Then exchange the ssh key from inside of one GraphStorm Docker containers to the rest containers in the cluster, i.e., copy the keys from the ``/root/.ssh/id_rsa.pub`` from one container to ``/root/.ssh/authorized_keys`` in containers on all other containers.
+
+.. note::
+
+    If you are planning to use **parmetis** algorithm, please prepare your docker container by instructions here:
+
+    .. code-block:: bash
+
+        git clone https://github.com/awslabs/graphstorm.git
+
+        cd /path-to-graphstorm/docker/
+
+        bash /path-to-graphstorm/docker/build_docker_parmetis.sh /path-to-graphstorm/ image-name image-tag
+
+    There are four positional arguments for ``build_docker_parmetis.sh``:
+
+    1. **path-to-graphstorm** (**required**), is the absolute path of the "graphstorm" folder, where you cloned the GraphStorm source code. For example, the path could be ``/code/graphstorm``.
+    2. **image-name** (optional), is the assigned name of the to be built Docker image. Default is ``graphstorm``.
+    3. **image-tag** (optional), is the assigned tag prefix of the Docker image. Default is ``local``.
+
+    The other stuff should remain the same.
```

Comment:
What are the "the other stuff"? If no other stuff, may remove this sentence.

Reply:
I mean the "file system & instances setup" should remain the same.

Reply:
I delete the sentence finally, I find it useless.